tools/lsmcli/cmdline.py

```diff
@@ -1597,21 +1625,40 @@ def process(self, cli=None):
         """
         Process the parsed command.
         """
-        if cli:
-            #Directly invoking code though a wrapper to catch unsupported
-            #operations.
-            self.c = Proxy(cli())
-            self.c.plugin_register(self.uri, self.password, self.tmo)
-            self.cleanup = self.c.plugin_unregister
+        if self.is_connection_free_cmd():
+            self.args.func(self.args)
         else:
-            #Going across the ipc pipe
-            self.c = Proxy(Client(self.uri, self.password, self.tmo))
-
-            if os.getenv('LSM_DEBUG_PLUGIN'):
-                raw_input(
-                    "Attach debugger to plug-in, press <return> when ready...")
-
-            self.cleanup = self.c.close
-
-        self.args.func(self.args)
-        self.shutdown()
+            if cli:
+                #Directly invoking code though a wrapper to catch unsupported
+                #operations.
+                self.c = Proxy(cli())
+                self.c.plugin_register(self.uri, self.password, self.tmo)
+                self.cleanup = self.c.plugin_unregister
+            else:
+                #Going across the ipc pipe
+                self.c = Proxy(Client(self.uri, self.password, self.tmo))
+
+                if os.getenv('LSM_DEBUG_PLUGIN'):
+                    raw_input("Attach debugger to plug-in, "
+                              "press <return> when ready...")
+
+                self.cleanup = self.c.close
+
+            self.args.func(self.args)
+            self.shutdown()
+
+    def local_disk_list(self, args):
+        local_disks = []
+        for sd_name in os.listdir("/sys/block/"):
+            if not sd_name.startswith("sd"):
+                continue
+            sd_path = "/dev/%s" % sd_name
+            if not os.path.exists(sd_path):
+                continue
+            vpd83 = LocalDisk.vpd83_get(sd_path)
+            rpm = LocalDisk.rpm_get(sd_path)
+            link_type = LocalDisk.link_type_get(sd_path)
+            local_disks.append(
+                LocalDiskInfo(sd_path, vpd83, rpm, link_type))
```

Comment:
This _looks_ like an opportunity for pyudev, something like:

```
for dev in context.list_devices(subsystem="block").match_is_initialized():
    if not dev.sys_name.startswith("sd"): # There might be a better way to check this
        continue
    sd_path = dev.device_node
    vpd83... # no need to check existance of device node, is_initialized() evidently takes care of that
```
and so forth as before.

But, you have to get the context somewhere, like: ```context = Context()```